### PR TITLE
Button mixin and documentation update

### PR DESCRIPTION
**Pull request by mturnwall**
At first I was trying to figure out how to make it more configurable and have hover states, but I decided to keep it as simple as possible for now.

Also added some quick documentation on how to use mixins since there was an issue with compiling. I documented the solution so no one else has the same issue.

### CODE REVIEW

File: styles/mixins/button.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+module.exports = function () {
+  return function (mixin, bgColor, color) {
```
Comment (jgallen23) [inline styles/mixins/button.js]: we also need a outline/hollow button. should this be part of this mixin or separate?  I guess we could just create a mixin that calls this mixing with a  border-color and then set bgColor to transparent.  thoughts?

File: readme.md [hunk 2]
```diff
@@ -8,3 +8,18 @@ When working on clientkit make sure you have the following tools installed:
    - Nodenv or similar (should match the version in .node-version)
 
 Project has been setup to be run and debug from inside VS Code. Alternately you can run `npm start`.
+
+### Using Mixins
```
Comment (jgallen23) [inline readme.md]: lets move this into a docs/mixins.md

File: styles/helpers/buttons.css [hunk 3]
```diff
@@ -0,0 +1,5 @@
+.btn-primary {
+  @mixin button var(--color-accent), var(--color-link);
+  width: 300px;
+  padding: 10px 0;
```
Comment (jgallen23) [inline styles/helpers/buttons.css]: I think padding should be part of mixin, you?

File: styles/mixins/button.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+    return {
+      'background-color': bgColor,
+      color,
+      display: 'inline-block',
+      'line-height': 1
```
Comment (jgallen23) [inline styles/mixins/button.js]: don't we need text-align: center too?

Comment (jgallen23) [inline styles/mixins/button.js]: doesn't line height need to be the height of the button?

Comment (jgallen23) [inline styles/mixins/button.js]: probably want to pass in height and then default it to 25px or so

File: styles/helpers/buttons.css (hunk 3 above)
Comment (chrisopedia) [inline styles/helpers/buttons.css]: Should we be forcing a width?

Comment (jgallen23) [inline styles/helpers/buttons.css]: nope.  I think we should have left/right padding instead

File: styles/mixins/button.js (hunk 4 above)
Comment (chrisopedia) [inline styles/mixins/button.js]: Why not do padding on top and bottom. Then we can get different sizes

Comment (jgallen23) [inline styles/mixins/button.js]: yeah, that works for me.  we probably also need a reset -webkit-appearance too, right?

File: styles/helpers/buttons.css (hunk 3 above)
Comment (chrisopedia) [inline styles/helpers/buttons.css]: Might be nice to have the ability to pass in a `max-width`

File: styles/mixins/button.js (hunk 4 above)
Comment (chrisopedia) [inline styles/mixins/button.js]: I know it's gross, but I'd only do that to an attributed selected input element.

```
input[type="button"],
Input[type="submit"]
```

**Comment by mturnwall** [comment]
> webkit-appearance

Chris is right that that property should only be applied to button elements. Using attribute selectors is just fine.

> doesn't line height need to be the height of the button?

Setting the `line-height` to one helps to remove the differences on how browsers apply it. Using padding on top and bottom gives better consistency for sizing.

> we also need a outline/hollow button

Wouldn't the helper class take care of this? It would just be a class that calls the button mixin with a background color of transparent.

I made the mixin simple with the idea of helper classes adding the specifics.

> Should we be forcing a width?

This one I'm not sure about. As part of the design language will buttons always be the same size? Should the mixin be setting the width or should the site's theme set it? Will width's be changing at breakpoints? For example on desktop next and cancel buttons are next to each other but on mobile they span the full width stacked.

I added a `text-align` center since I don't think I've ever seen a button whose text wasn't centered aligned 😄

**Comment by dawnerd** [comment]
@mturnwall can you merge in master and fix whatever conflict there is?

**Pull request merged**